BigQuery/tests/Snippet/ModelTest.php

```diff
@@ -0,0 +1,140 @@
+    const MODEL_ID = 'myModelId';
+
+    public function setUp()
+    {
+        $this->connection = $this->prophesize(ConnectionInterface::class);
+    }
+
+    public function getModel($connection, array $info = [])
+    {
+        return new Model(
+            $connection->reveal(),
+            self::MODEL_ID,
+            self::DATASET_ID,
+            self::PROJECT_ID,
+            $info
+        );
+    }
+
+    public function testExists()
+    {
+        $this->connection->getModel(Argument::any())
+            ->shouldBeCalledTimes(1)
+            ->willReturn([]);
+        $model = $this->getModel($this->connection);
+        $snippet = $this->snippetFromMethod(Model::class, 'exists');
+        $snippet->addLocal('model', $model);
+        $res = $snippet->invoke('model');
+
+        $this->assertEquals(true, $res->output());
+    }
+
+    public function testDelete()
+    {
+        $this->connection->deleteModel(Argument::any())
+            ->shouldBeCalledTimes(1)
+            ->willReturn([]);
+
+        $model = $this->getModel($this->connection);
+        $snippet = $this->snippetFromMethod(Model::class, 'delete');
+        $snippet->addLocal('model', $model);
+
+        $snippet->invoke();
+    }
+
+    public function testInfo()
+    {
+        $this->connection->getModel(Argument::any())
+            ->shouldBeCalledTimes(1)
+            ->willReturn([
+                'modelType' => 'LOGISTIC_REGRESSION'
+            ]);
+
+        $model = $this->getModel($this->connection);
+        $snippet = $this->snippetFromMethod(Model::class, 'info');
+        $snippet->addLocal('model', $model);
+        $res = $snippet->invoke('model');
+
+        $this->assertEquals('LOGISTIC_REGRESSION', $res->output());
+    }
+
+    public function testReload()
+    {
+        $this->connection->getModel(Argument::any())
+            ->shouldBeCalledTimes(1)
+            ->willReturn([
+                'modelType' => 'LOGISTIC_REGRESSION'
+            ]);
+
+        $model = $this->getModel($this->connection);
+        $snippet = $this->snippetFromMethod(Model::class, 'reload');
+        $snippet->addLocal('model', $model);
+        $res = $snippet->invoke('model');
+
+        $this->assertEquals('LOGISTIC_REGRESSION', $res->output());
+    }
+
+    public function testId()
+    {
+        $model = $this->getModel($this->connection);
+        $snippet = $this->snippetFromMethod(Model::class, 'id');
+        $snippet->addLocal('model', $model);
+        $res = $snippet->invoke('model');
+
+        $this->assertEquals(self::MODEL_ID, $res->output());
+    }
+
+    public function testIdentity()
+    {
+        $model = $this->getModel($this->connection);
+        $snippet = $this->snippetFromMethod(Model::class, 'identity');
+        $snippet->addLocal('model', $model);
+        $res = $snippet->invoke('model');
+
+        $this->assertEquals(self::MODEL_ID, $res->output());
+    }
+
+    public function testUpdate()
```

Comment:
1. Let's please add an assertion to this test.
2. Add a similar test which ensures e-tags are handled correctly.